### PR TITLE
refactor(t2viz): register the AI actions to query controls in discover

### PR DESCRIPTION
**Pull request by ruanyl**
### Description
image
image
image

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test.
- [ ] New functionality has user manual doc added.
- [ ] Commits are signed per the DCO using --signoff.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: common/types/config.ts [hunk 1]
```diff
@@ -25,6 +25,9 @@ export const configSchema = schema.object({
   smartAnomalyDetector: schema.object({
     enabled: schema.boolean({ defaultValue: false }),
   }),
+  branding: schema.object({
+    label: schema.string({ defaultValue: 'AI assistant' }),
+  }),
 });
```
Comment (Hailong-am) [inline common/types/config.ts:31]: nit: do we need i18n here?

**Review by Hailong-am — APPROVED** [review]

File: public/components/ui_action_context_menu.tsx [hunk 2]
```diff
@@ -38,12 +42,20 @@ export const ActionContextMenu = () => {
   return (
     <EuiPopover
       button={
-        <EuiButtonIcon
+        <EuiButtonEmpty
+          color="text"
           aria-label="AI assistant trigger button"
-          size="s"
-          iconType={assistantTriggerIcon}
+          size="xs"
```
Comment (Hailong-am) [inline public/components/ui_action_context_menu.tsx:48]: maybe we should keep it as `s` to align with DQL

image

File: common/types/config.ts (hunk 1 above)
Comment (ruanyl) [reply common/types/config.ts:31]: This value will be from config yaml, I'm not sure if i18n will even work for config yaml

Comment (ruanyl) [reply common/types/config.ts:31]: You mean i18n for the default value?

Comment (Hailong-am) [reply common/types/config.ts:31]: > This value will be from config yaml, I'm not sure if i18n will even work for config yaml

I see, if the value from config yaml, the value should be whatever specified in yaml file. can we make default value as empty? that will use the fallback value that supports i18n

File: public/components/ui_action_context_menu.tsx (hunk 2 above)
Comment (ruanyl) [reply public/components/ui_action_context_menu.tsx:48]: having an alignment issue now when changed to a `s` button, this requires a css tweaks in core, I will keep as is now.
image

File: common/types/config.ts (hunk 1 above)
Comment (ruanyl) [reply common/types/config.ts:31]: Good point, I will update

**Pull request merged**